fragments/rhn-register.sh

```diff
@@ -15,17 +21,52 @@ set -x
 # Return the final non-zero exit code of a failed pipe (or 0 for success)
 set -o pipefail
 
+function use_satellite6() {
+		[ -n "$SAT6_HOSTNAME" ]
+}
+
+function use_rhn() {
+		[ -n "$RHN_USERNAME" -a -n "$RHN_PASSWORD" ]
+}
+
+function register_rhn() {
+		# RHN_USERNAME=$1
+		# RHN_PASSWORD=$2
+		retry subscription-manager register --username="$1" --password="$2"
+}
+
+function install_sat6_ca_certs() {
+		# SAT6_HOSTNAME=$1
+		local SAT6_KEY_RPM="katello-ca-consumer-$1"
+		local SAT6_KEY_RPM_URL="$https://${1}/pub/katello-ca-consumer-latest.noarch.rpm"
+
+		if ! rpm -q --quiet $SAT6_KEY_RPM
+		then
+				yum -y install $SAT6_KEY_RPM_URL
+		fi
+}
+
+function register_sat6() {
+		# SAT6_ORGANIZATION=$1
+		# SAT6_ACTIVATIONKEY=$2
+		# register as a sat6 client
+		retry subscription-manager regiester --org="$1" --activationkey="$1"
```

Comment:
um, I'm not sure if this code was really tested - "regiester" would probably never pass